src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -95,7 +95,20 @@ public static StringComparer Create(CultureInfo culture, bool ignoreCase)
                 throw new ArgumentNullException(nameof(culture));
             }
 
-            return new CultureAwareComparer(culture, ignoreCase);
+            if (ignoreCase)
+                return new CultureAwareComparer(culture, CompareOptions.IgnoreCase);
+
+            return new CultureAwareComparer(culture, CompareOptions.None);
```

Comment:
nit: maybe you can do it in one line using ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;